pkg/util/ciutil/vars.go

```diff
@@ -43,6 +43,19 @@ func DetectVars() Vars {
 	// try to detect some additional CI-specific metadata that the CLI will use. It's okay if we can't,
 	// we'll just have reduced functionality for our various CI integrations.
 	switch v.Name {
+	case AzurePipelines:
+		// See: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables
+		v.BuildID = os.Getenv("Build.BuildId")
```

Comment:
I remember reading [this](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=vsts&tabs=yaml%2Cbatch#set-in-script) note about accessing variables in scripts a while ago. I am not sure if that applies here? Meaning, should we access those variables as `BUILD_REASON` etc.? I know that if you are referencing these variables within the Azure Pipeline you would use the `.` syntax, but I am unsure otherwise.

I'd be curious to see if a dev build of the CLI with this change picks up the variable regardless of the Agent OS setting of the Azure pipeline job.

Reply:
Thanks for pointing that out, yeah, it does sound like more configuration is needed. I'll try to setup Azure Pipelines in our Azure account to test this.

Reply:
They should be converted to the X_Y style yeah - TF_BUILD is "True" when running in an Azure pipeline.